base/shell/explorer/traywnd.cpp

```diff
@@ -3348,27 +3361,15 @@ class CTrayWindow :
         ::SendMessageW(m_TrayNotify, uMsg, wParam, lParam);
 
         /* Toggle autohide */
-        if (newSettings->sr.AutoHide != g_TaskbarSettings.sr.AutoHide)
-        {
-            g_TaskbarSettings.sr.AutoHide = newSettings->sr.AutoHide;
-            memset(&m_AutoHideOffset, 0, sizeof(m_AutoHideOffset));
-            m_AutoHideState = AUTOHIDE_SHOWN;
-            if (!newSettings->sr.AutoHide)
-                SetWindowPos(NULL, 0, 0, 0, 0, SWP_NOACTIVATE | SWP_NOZORDER);
-            else
-                SetTimer(TIMER_ID_MOUSETRACK, MOUSETRACK_INTERVAL, NULL);
-        }
+        if (newSettings->sr.AutoHide != IsAutoHideState())
+            SetAutoHideState(newSettings->sr.AutoHide);
 
         /* Toggle lock state */
         Lock(newSettings->bLock);
 
         /* Toggle OnTop state */
         if (newSettings->sr.AlwaysOnTop != g_TaskbarSettings.sr.AlwaysOnTop)
```

Comment:
Would it make sense to do this test (plus do the update only if the values are different) within the UpdateAlwaysOnTop function itself? Or would this collide with what's intended for the call made in `CAppBarManager::OnAppBarSetState` in appbar.cpp ?

Reply:
I made it unconditional.

Reply:
> would this collide with what's intended for the call made in CAppBarManager::OnAppBarSetState in appbar.cpp ?

Nope.